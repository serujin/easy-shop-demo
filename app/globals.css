@import url("https://p.typekit.net/p.css?s=1&k=gws6bus&ht=tk&f=139&a=10612664&app=typekit&e=css");
@font-face {
  font-family: "proxima-nova";
  src: url("https://use.typekit.net/af/949f99/00000000000000003b9b3068/27/l?primer=7cdcb44be4a7db8877ffa5c0007b8dd865b3bbc383831fe2ea177f62257a9191&fvd=n7&v=3")
      format("woff2"),
    url("https://use.typekit.net/af/949f99/00000000000000003b9b3068/27/d?primer=7cdcb44be4a7db8877ffa5c0007b8dd865b3bbc383831fe2ea177f62257a9191&fvd=n7&v=3")
      format("woff"),
    url("https://use.typekit.net/af/949f99/00000000000000003b9b3068/27/a?primer=7cdcb44be4a7db8877ffa5c0007b8dd865b3bbc383831fe2ea177f62257a9191&fvd=n7&v=3")
      format("opentype");
  font-display: auto;
  font-style: normal;
  font-weight: 700;
  font-stretch: normal;
}

* {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
  font-family: "proxima-nova";
}

html {
  color: #404040;
  font-size: 14px;
  background-color: #efefef;
}

body {
  display: flex;
  flex-direction: column;
}

header {
  display: flex;
  background-color: #4e2817;
  justify-content: center;
}

a {
  text-decoration: none;
  color: black;
}
