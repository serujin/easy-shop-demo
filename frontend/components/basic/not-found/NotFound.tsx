import Ratings from '@/components/ratings/Ratings'
import customStickersStyles from '@/styles/custom-stickers-page.module.css'
import styles from './NotFound.module.css'

type NotFoundProps = {
  isIndex?: boolean
}

const getHeader = (isIndex: boolean) => {
  if (!isIndex) {
    return
  }
  return (
    <div className={customStickersStyles.headerContainer}>
      <div className={customStickersStyles.ratingsContainer}>
        <Ratings
          title={'Custom stickers'}
          titleStyle="huge"
          rating={4.8}
          reviews={155761}
        />
      </div>
    </div>
  )
}

const getNotFoundText = (isIndex: boolean) => {
  const pageText = isIndex ? 'product' : 'product information'
  return `We could not load any ${pageText} right now, try again later`
}

export default function NotFound(props: NotFoundProps) {
  const { isIndex = false } = props
  const containerClassName = isIndex
    ? styles.notFoundIndexContainer
    : styles.notFoundProductContainer
  return (
    <main className={customStickersStyles.main}>
      {getHeader(isIndex)}
      <div className={containerClassName}>
        <p className={styles.notFoundText}>{getNotFoundText(isIndex)}</p>
        <p className={styles.notFoundText}>Sorry for the inconveniences</p>
      </div>
    </main>
  )
}
