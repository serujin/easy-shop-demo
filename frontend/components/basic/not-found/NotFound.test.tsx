import React from 'react'
import { render } from '@testing-library/react'
import { describe, expect, it } from '@jest/globals'
import NotFound from './NotFound'

describe('NotFound component', () => {
  it('should render correctly without props', () => {
    const { container } = render(<NotFound />)
    expect(container).toMatchSnapshot()
  })
  it('should render correctly with isIndex prop', () => {
    const { container } = render(<NotFound isIndex />)
    expect(container).toMatchSnapshot()
  })
  it('should render correctly with is404 prop', () => {
    const { container } = render(<NotFound is404 />)
    expect(container).toMatchSnapshot()
  })
})
