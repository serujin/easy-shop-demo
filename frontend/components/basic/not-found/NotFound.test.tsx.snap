// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound component should render correctly with is404 prop 1`] = `
<div>
  <main
    class="main"
  >
    <div
      class="notFoundProductContainer"
    >
      <p
        class="notFoundText"
      >
        Oops, the page you are trying to reach doesn't exist
      </p>
      <p
        class="notFoundText"
      >
        Sorry for the inconveniences
      </p>
    </div>
  </main>
</div>
`;

exports[`NotFound component should render correctly with isIndex prop 1`] = `
<div>
  <main
    class="main"
  >
    <div
      class="headerContainer"
    >
      <div
        class="ratingsContainer"
      >
        <div
          class="container"
        >
          <div
            class="titleContainer"
          >
            <p
              class="huge"
            >
              Custom stickers
            </p>
          </div>
          <div
            class="reviewsContainer huge"
          >
            <div
              class="starsContainer"
            >
              <img
                alt="Full star"
                class="starItem"
                data-nimg="1"
                decoding="async"
                fetchpriority="high"
                height="17"
                src="/ratings/full-star.svg"
                style="color: transparent;"
                width="17"
              />
              <img
                alt="Full star"
                class="starItem"
                data-nimg="1"
                decoding="async"
                fetchpriority="high"
                height="17"
                src="/ratings/full-star.svg"
                style="color: transparent;"
                width="17"
              />
              <img
                alt="Full star"
                class="starItem"
                data-nimg="1"
                decoding="async"
                fetchpriority="high"
                height="17"
                src="/ratings/full-star.svg"
                style="color: transparent;"
                width="17"
              />
              <img
                alt="Full star"
                class="starItem"
                data-nimg="1"
                decoding="async"
                fetchpriority="high"
                height="17"
                src="/ratings/full-star.svg"
                style="color: transparent;"
                width="17"
              />
              <img
                alt="Full star"
                class="starItem"
                data-nimg="1"
                decoding="async"
                fetchpriority="high"
                height="17"
                src="/ratings/full-star.svg"
                style="color: transparent;"
                width="17"
              />
            </div>
            <p>
              155,761 reviews
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="notFoundIndexContainer"
    >
      <p
        class="notFoundText"
      >
        We could not load any product right now, try again later
      </p>
      <p
        class="notFoundText"
      >
        Sorry for the inconveniences
      </p>
    </div>
  </main>
</div>
`;

exports[`NotFound component should render correctly without props 1`] = `
<div>
  <main
    class="main"
  >
    <div
      class="notFoundProductContainer"
    >
      <p
        class="notFoundText"
      >
        We could not load any product information right now, try again later
      </p>
      <p
        class="notFoundText"
      >
        Sorry for the inconveniences
      </p>
    </div>
  </main>
</div>
`;
